File: hackerrank/30 days of code/day 19, interfaces/day_19_interfaces.py
class AdvancedArithmetic(object):
    def divisorSum(n):
        raise NotImplementedError


class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        sum = 0
        for i in range(1, int(n / 2) + 1):
            if not (n % i):
                sum += i
        return sum + n


# Imutable HackerRank code below
n = int(input())
my_calculator = Calculator()
s = my_calculator.divisorSum(n)
print("I implemented: " + type(my_calculator).__bases__[0].__name__)
print(s)

File: hackerrank/30 days of code/day 19, interfaces/day_19_interfaces.ts
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');
let inputString: string = '';
let inputLines: string[] = [];
let currentLine: number = 0;
process.stdin.on('data', function(inputStdin: string): void {
    inputString += inputStdin;
});

process.stdin.on('end', function(): void {
    inputLines = inputString.split('\n');
    inputString = '';
    main();
});

function readLine(): string {
    return inputLines[currentLine++];
}

interface AdvancedArithmetic {
    divisorSum(n: number) : number
}

class Calculator implements AdvancedArithmetic {
    divisorSum(n: number) : number {
        let sum = 0;
        for (let i = 1; i <= n / 2; i++)
            if (!(n % i))
                sum += i;
        return sum + n;
    }
}

function main() {
    let n = +readLine()
    let calc = new Calculator()
    console.log(`I implemented: AdvancedArithmetic \n${calc.divisorSum(n)}`)
}